[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-cookie

Disallow direct access to document.cookie. Use mw.cookie instead.

üìã This rule is enabled in `plugin:mediawiki/common`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
document.cookie = 'foo=bar';
console.log( document.cookie );
window.document.cookie = 'foo=bar';
console.log( window.document.cookie );
```

‚úîÔ∏è Examples of **correct** code:
```js
mw.cookie.get( 'foo' );
mw.cookie.set( 'bar', 'baz' );
x = cookie;
```

## Resources

* [Rule source](/src/rules/no-cookie.js)
* [Test source](/tests/rules/no-cookie.js)
