[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-extended-unicode-identifiers

Prohibits non-BMP characters in identifiers, for Safari 11-13 compatibility.

## Rule details

❌ Examples of **incorrect** code:
```js
var 𐋀 = 13;
const 𠁂 = 13;
```

✔️ Examples of **correct** code:
```js
var x = 42;
const ꭳ = 42;
const Ⲃ = 42;
```

## Resources

* [Rule source](/src/rules/no-extended-unicode-identifiers.js)
* [Test source](/tests/rules/no-extended-unicode-identifiers.js)
