[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-extended-unicode-identifiers

Prohibits non-BMP characters in identifiers, for Safari 11-13 compatibility.

## Rule details

âŒ Examples of **incorrect** code:
```js
var ğ‹€ = 13;
const ğ ‚ = 13;
```

âœ”ï¸ Examples of **correct** code:
```js
var x = 42;
const ê­³ = 42;
const â²‚ = 42;
```

## Resources

* [Rule source](/src/rules/no-extended-unicode-identifiers.js)
* [Test source](/tests/rules/no-extended-unicode-identifiers.js)
