[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-storage

Disallow direct access to localStorage and sessionStorage. Use mw.storage or mw.storage.session instead.

üìã This rule is enabled in `plugin:mediawiki/common`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
localStorage.getItem( 'foo' );
sessionStorage.setItem( 'bar', 'baz' );
window.localStorage.getItem( 'foo' );
window.sessionStorage.setItem( 'bar', 'baz' );
```

‚úîÔ∏è Examples of **correct** code:
```js
mw.storage.get( 'foo' );
mw.storage.session.get( 'bar' );

function foo( localStorage ) {
    return localStorage.getItem( 'foo' );
}
```

## Resources

* [Rule source](/src/rules/no-storage.js)
* [Test source](/tests/rules/no-storage.js)
