[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-unlabeled-buttonwidget

Warn when a label-less OO.ui.ButtonWidget is created without a label or invisible label.

üìã This rule is enabled in `plugin:mediawiki/common`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
widget = new OO.ui.ButtonWidget( { icon: 'edit' } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', label: '' } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', title: 'Title' } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', invisibleLabel: false } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', invisibleLabel: true } );
widget = new OO.ui.ButtonWidget( { [ computedProbablyNotLabel ]: 'Some value' } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', ...label } );
```

‚úîÔ∏è Examples of **correct** code:
```js
widget = new OO.ui.IconWidget( { icon: 'edit' } );
widget = new OO.ui.ButtonWidget( { label: 'Label' } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', label: 'Label' } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', invisibleLabel: true, label: 'Label' } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', invisibleLabel: false, label: 'Label' } );
widget = new OO.ui.ButtonWidget( { icon: 'edit', label() {} } );
```

## Resources

* [Rule source](/src/rules/no-unlabeled-buttonwidget.js)
* [Test source](/tests/rules/no-unlabeled-buttonwidget.js)
