[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# storage-expiry

Require expiry when using mw.storage with a non-literal key.

ğŸ“‹ This rule is enabled in `plugin:mediawiki/common`.

## Rule details

âŒ Examples of **incorrect** code:
```js
mw.storage( variable, value );
mw.storage( variable, value, undefined );
```

âœ”ï¸ Examples of **correct** code:
```js
mw.storage( 'key', value );
mw.storage( 'key', value, 3600 );
mw.storage( variable, value, 3600 );
mw.storage( variable, value, expiry );
value = mw.storage( 'key' );
value = mw.storage( variable );
```

## Resources

* [Rule source](/src/rules/storage-expiry.js)
* [Test source](/tests/rules/storage-expiry.js)
