[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# vue-exports-component-directive

Require `defineComponent()` calls or `// @vue/component` directives to trigger eslint-plugin-vue rules

üìã This rule is enabled in `plugin:mediawiki/vue`.

üîß The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

‚ùå Examples of **incorrect** code:
```vue
<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
// @vue/component

module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports = {};
// @vue/component
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports = notDefineComponent( {} );
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
const { defineComponent } = require( 'vue' );
module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
const { ref } = require( 'vue' );
module.exports = {};
</script>
```

‚úîÔ∏è Examples of **correct** code:
```vue
<template>
    <p>Placeholder...</p>
</template>
<script>
// @vue/component
module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
/* @vue/component */
module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports = defineComponent( {} );
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
// @vue/component
module.exports = exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports = exports = defineComponent( {} );
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports += 5;
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.imports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
foo.exports = {};
</script>
```

üîß Examples of code **fixed** by this rule:
```vue
<template>                                    /* ‚Üí */ <template>
    <p>Placeholder...</p>                     /* ‚Üí */     <p>Placeholder...</p>
</template>                                   /* ‚Üí */ </template>
<script>                                      /* ‚Üí */ <script>
module.exports = {};                          /* ‚Üí */ const { defineComponent } = require( 'vue' );
</script>;                                    /* ‚Üí */ module.exports = defineComponent( {} );
                                              /* ‚Üí */ </script>;

<template>                                    /* ‚Üí */ <template>
    <p>Placeholder...</p>                     /* ‚Üí */     <p>Placeholder...</p>
</template>                                   /* ‚Üí */ </template>
<script>                                      /* ‚Üí */ <script>
// @vue/component                             /* ‚Üí */ // @vue/component
                                              /* ‚Üí */
module.exports = {};                          /* ‚Üí */ const { defineComponent } = require( 'vue' );
</script>;                                    /* ‚Üí */ module.exports = defineComponent( {} );
                                              /* ‚Üí */ </script>;

<template>                                    /* ‚Üí */ <template>
    <p>Placeholder...</p>                     /* ‚Üí */     <p>Placeholder...</p>
</template>                                   /* ‚Üí */ </template>
<script>                                      /* ‚Üí */ <script>
module.exports = {};                          /* ‚Üí */ const { defineComponent } = require( 'vue' );
// @vue/component                             /* ‚Üí */ module.exports = defineComponent( {} );
</script>;                                    /* ‚Üí */ // @vue/component
                                              /* ‚Üí */ </script>;

<template>                                    /* ‚Üí */ <template>
    <p>Placeholder...</p>                     /* ‚Üí */     <p>Placeholder...</p>
</template>                                   /* ‚Üí */ </template>
<script>                                      /* ‚Üí */ <script>
const { defineComponent } = require( 'vue' ); /* ‚Üí */ const { defineComponent } = require( 'vue' );
module.exports = {};                          /* ‚Üí */ module.exports = defineComponent( {} );
</script>;                                    /* ‚Üí */ </script>;

<template>                                    /* ‚Üí */ <template>
    <p>Placeholder...</p>                     /* ‚Üí */     <p>Placeholder...</p>
</template>                                   /* ‚Üí */ </template>
<script>                                      /* ‚Üí */ <script>
const { ref } = require( 'vue' );             /* ‚Üí */ const { ref, defineComponent } = require( 'vue' );
module.exports = {};                          /* ‚Üí */ module.exports = defineComponent( {} );
</script>                                     /* ‚Üí */ </script>
```

## Resources

* [Rule source](/src/rules/vue-exports-component-directive.js)
* [Test source](/tests/rules/vue-exports-component-directive.js)
