[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# vue-exports-component-directive

Require `// @vue/component` directives to trigger eslint-plugin-vue rules

üìã This rule is enabled in `plugin:mediawiki/vue`.

## Rule details

‚ùå Examples of **incorrect** code:
```vue
<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
// @vue/component

module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports = {};
// @vue/component
</script>
```

‚úîÔ∏è Examples of **correct** code:
```vue
<template>
    <p>Placeholder...</p>
</template>
<script>
// @vue/component
module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
/* @vue/component */
module.exports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.exports += 5;
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
module.imports = {};
</script>;

<template>
    <p>Placeholder...</p>
</template>
<script>
foo.exports = {};
</script>
```

## Resources

* [Rule source](/src/rules/vue-exports-component-directive.js)
* [Test source](/tests/rules/vue-exports-component-directive.js)
